CONTRIBUTING.md

```diff
@@ -468,15 +456,22 @@ ensure that any changes which impact the rules of the specification (including,
 but not limited to, adding new entities, suffixes, datatypes, modalities) are
 reflected in the schema as well.
 
-#### 2. Generate an updated entity table before pushing your changes
+#### 2. Ensure that changes to the schema are matched in auto-generated sections of the specification
 
-Run the Python script `tools/bids_schema.py`:
+The schema is used to generate a number of elements in the specification text, including:
+- Filename format templates
+- Entity tables
+- Entity definitions
 
-```bash
-python tools/bids_schema.py entity src/schema/ src/99-appendices/04-entity-table.md
-```
+As such, you need to ensure that the functions used throughout the specific to render these elements are appropriately referencing the schema; in essence, that, if your changes do impact how functions should be called, the function calls have been updated.
+
+#### 3. Render the specification with `mkdocs` to check your changes
+
+Run `mkdocs serve` and open `localhost:8000` to browse the rendered specification.
+Make sure that all filename format templates, entity tables, and entity definitions are correct
+and that the code that generates these elements is not broken by your changes.
```

Comment:
You're right, the CI will render the files, but the CI _won't_ fail if the code fails.

Reply:
oh, how come is that?  because it is too deep (invoked through remake etc) and something just swallows its failures?  Ideally we should fail if anything fails in the code... if not possible "legitimately", could be done by wrapping `__main__` invocation in try/except and then creating some sentinel file and then checking after invocation of the outside tool that there is no such file.

just a comment/idea: should not stop this PR from being merged!

Reply:
I think it's a problem with the `mkdocs-macros` plugin.